app/api/routes/user.py

```diff
@@ -37,3 +47,166 @@ async def create_user(user: UserCreateRequestModel):
+        user.hashed_password = get_password_hash(update_data.password)
+
+    user.updated_at = datetime.now(timezone.utc)
+
+    users_db[str(user.id)] = user
+
+    return UserUpdateResponseModel(
+        id=user.id,
+        username=user.username,
+        email=user.email,
+        is_admin=user.is_admin,
+        is_active=user.is_active,
+        created_at=user.created_at,
+        updated_at=user.updated_at
+    )
+
+
+
+#DELETE User
+@router.delete("/users/{user_id}", status_code=status.HTTP_204_NO_CONTENT)
+async def delete_user(user_id: UUID, current_user: User = Depends(get_current_user)):
+    
+    if str(user_id) != str(current_user.id):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="You can only delete your own account.")
+
+    user = users_db.get(str(user_id))
+    if not user:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found.")
+
+    """
+    This part will be for checking active orders
+    """
+
+    del users_db[str(user_id)]
+    #Return 204 no content
+    return
+
+
+#GET all users 
+@router.get("/users", response_model=list[UserCreateResponseModel], status_code=status.HTTP_200_OK)
+async def get_all_users(current_admin: User = Depends(get_current_admin)):
+
+    try:
+        all_users = list(users_db.values())
+
+        return [
+            UserUpdateResponseModel(
+                id=user.id,
+                username=user.username,
+                email=user.email,
+                is_admin=user.is_admin,
+                is_active=user.is_active,
+                created_at=user.created_at,
+                updated_at=user.updated_at
+            )
+            for user in all_users
+        ]
+
+    except Exception as e:
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail=f"An error occurred while retrieving users: {str(e)}"
+        )
+    
+
+
+#List orders for a user 
+@router.get("/users/{user_id}/orders", status_code=status.HTTP_200_OK)
+async def list_user_orders(user_id: UUID, current_user: User = Depends(get_current_active_user)):
+    
+    if str(user_id) != str(current_user.id):
+        raise HTTPException(status_code=status.HTTP_403_FORBIDDEN, detail="You can only view your own orders.")
+    
+    """
+        Code in progress
+
+    """
+
+
+#Change role 
+@router.put("/users/change_role", status_code=status.HTTP_200_OK)
+async def change_role(request: ChangeRoleRequest, current_admin: User = Depends(get_current_admin)):
+
+    try:
+        user = users_db.get(str(request.user_id))
+        if not user:
+            raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail="User not found.")
+
+        user.is_admin = request.is_admin
+        users_db[str(request.user_id)] = user  
+
+        return {"message": "User role updated successfully."}
+    
+    except HTTPException as http_exc:
+        raise http_exc
```

Comment:
What happens here? You raise an HTTPException, then catch it, and then raise it again? Why?

Reply:
It's my fault, see if it's working good now.